src/LoginServer/Network/LoginPacketHandler.cs

```diff
@@ -415,6 +415,16 @@ public void CB_CHECK_CLIENT_INTEGRITY(LoginConnection conn, Packet packet)
 			if (checksum != result)
 			{
 				Send.BC_MESSAGE(conn, MsgType.InvalidIpf);
+
+				// Even though the integrity check fails, send these packets as well.
+				// This is done because the client is in a hanging state waiting for these packets.
+				// Without these, the invalid IPF message will not be visible.
+				Send.BC_COMMANDER_LIST(conn);
+				Send.BC_NORMAL_ZoneTraffic(conn);
+				Send.BC_NORMAL_TeamUI(conn);
+
+				// Terminate any further requests from the client at this point.
+				conn.IgnorePackets = true;
```

Comment:
I added this after finding out that the IPF mismatch message can be easily bypassed. Unfortunately, this check cannot be done until *after* the client actually enters the barrack, so there has to be something done about preventing the client from making further requests.